internal/kafka/internal/workers/cluster_mgrs/cleanup_cluster_external_resources_mgr.go

```diff
@@ -0,0 +1,145 @@
+package cluster_mgrs
+
+import (
+	fleeterrors "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/sso"
+
+	"sync"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/workers"
+	"github.com/goava/di"
+	"github.com/google/uuid"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/metrics"
+	"github.com/golang/glog"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	cleanupClusterExternalResourcesWorkerType = "cleanup_cluster_external_resources"
+)
+
+// CleanupClusterExternalResourcesManager represents a worker that periodically reconciles data plane clusters in cleanup state.
+type CleanupClusterExternalResourcesManager struct {
+	id           string
+	workerType   string
+	isRunning    bool
+	imStop       chan struct{} //a chan used only for cancellation.
+	syncTeardown sync.WaitGroup
+	CleanupClusterExternalResourcesManagerOptions
+}
+
+type CleanupClusterExternalResourcesManagerOptions struct {
```

Comment:
I personally prefer explicitly passing the dependencies individually in the constructor to be aware of them instead of using `di.Inject` in a struct (as it's done in the different kafka managers) , but that's a personal style preference.

Reply:
I can change it.  The reason I used it here was mostly because the two added workers were copy-pasted and since the number of arguments for the new worker was quite big as well I thought we can continue to keep them in the arg containter

Reply:
adjusted